### ISSUE TITLE
dynamic vverbose rendering should use process name

### ISSUE THREAD

**Issue by mr-tz** [mandiant/capa#1816]
this is much clearer than for example:

```
thread @ process ppid: 1264, process pid: 1312, thread id: 2544 in process process ppid: 1264, process pid: 1312
```

**Comment by williballenthin** [comment]
today:

![image](https://github.com/mandiant/capa/assets/156560/6d36c4c8-b92f-4c70-bb30-3d082d915514)

some things of note:
  - `process ppid: 2456, process pid: 3052, thread id: 2792, call: 828` doesn't really have much meaning to a human. would be nicer to see:
    - process name instead of/with PID, and
    - rendered API call instead of/with call index
  - there are many call entries with the same thread, so most of the metadata is repeated multiple times. would be better to show pid, tid once and only the differing call data.

**Comment by williballenthin** [comment]
```yaml
write file on Windows
namespace  host-interaction/file-system/write
author     william.ballenthin@mandiant.com, anushka.virgaonkar@mandiant.com
scope      Scope.PROCESS
mbc        File System::Writes File [C0052]
Scope.PROCESS @ process ppid: 2456, process pid: 3052
  or:
    and:
      os: windows
      optional:
        or:
          thread:
            or:
              number: 0x2 = FILE_WRITE_DATA @ process ppid: 2456, process pid: 3052, thread id: 2792, call: 828, process ppid: 2456, process pid: 3052, thread id: 2792, call: 834, process ppid: 2456, process pid: 3052, thread id: 2792, call: 850, process ppid: 2456, process pid: 3052, thread id: 2792, call: 866
```

```yaml
write file on Windows
namespace  host-interaction/file-system/write
author     william.ballenthin@mandiant.com, anushka.virgaonkar@mandiant.com
scope      Scope.PROCESS
mbc        File System::Writes File [C0052]
Scope.PROCESS @ process ppid: 2456, process pid: 3052
  or:
    and:
      os: windows
      optional:
        or:
          thread:
            or:
              number: 0x2 = FILE_WRITE_DATA @ explorer.exe[3052:2792], WriteFile[828](foo, bar), WriteFile[834](foo, baz), WriteFile[850](foo, woo), WriteFile[866](foo, hoo)
```

**Comment by mr-tz** [comment]
check https://github.com/mandiant/capa-testfiles/blob/dynamic-feature-extractor/dynamic/cape/v2.4/capesandbox_com/0a0882b8da225406cc838991b5f67d11.json.gz as a potential good test candidate (with many processes and repeated calls)

![image](https://github.com/mandiant/capa/assets/17606537/0cdf0170-4271-4474-98ca-e56a8062b1a4)

**Comment by mr-tz** [comment]
On first glance I like proposal 2 the most.

**Comment by williballenthin** [comment]
work in progress:

![Screenshot 2023-10-20 4 23 11 PM](https://github.com/mandiant/capa/assets/156560/6cce4036-973e-4ab7-ab09-6b4e06d636aa)

**Comment by williballenthin** [comment]
closed in #1825
